api/grpc_server.go

```diff
@@ -21,15 +22,74 @@ import (
 
 // SpacemeshGrpcService is a grpc server providing the Spacemesh api
 type SpacemeshGrpcService struct {
-	Server   *grpc.Server
-	Port     uint
-	StateApi StateAPI
-	Network  NetworkAPI
-	Tx       TxAPI
-	Mining   MiningAPI
-	Oracle   OracleAPI
-	GenTime  GenesisTimeAPI
-	Logging  LoggingAPI
+	Server        *grpc.Server
+	Port          uint
+	StateApi      StateAPI         // State DB
+	Network       NetworkAPI       // P2P Swarm
+	Tx            TxAPI            // Mesh
+	TxMempool     *miner.TxMempool // TX Mempool
+	Mining        MiningAPI        // ATX Builder
+	Oracle        OracleAPI
+	GenTime       GenesisTimeAPI
+	LayerDuration time.Duration
+	Logging       LoggingAPI
+}
+
+func (s SpacemeshGrpcService) getTransaction(txId types.TransactionId) (*types.Transaction, *types.LayerID, pb.TxStatus, error) {
+	tx, err := s.Tx.GetTransaction(txId) // have we seen this transaction in a block?
```

Comment:
IMO, this logic seems to belong inside as part of the MVC pattern. the grpc is merley the view builder

Reply:
Hmm... I see what you mean, but there's no place that currently has access to all the databases required. The mesh has no knowledge of the mempool and vice-versa.

I could create a transactions controller or something, but it feels like overkill for this little logic.

WDYT?